### PR TITLE
Fix bug (#251)

### PR DESCRIPTION
**Pull request by birdmanmandbir**
Solve issue #251 and add corresponding test for it
Scope annotation in ProvidedBy handling should be added before create provided by binding

### CODE REVIEW

File: core/test/com/google/inject/ProvidedByAndScopeTest.java [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+    public MixClass get() {
+      return new MixClass();
+    }
+  }
+
+  @Singleton
+  @ProvidedBy(MixClassProvider.class)
+  static class MixClass {
+    @Override
+    public boolean equals(Object obj) {
```
Comment (ogregoire) [inline core/test/com/google/inject/ProvidedByAndScopeTest.java:36]: This method is useless as it's the default behavior of Object.

File: core/test/com/google/inject/ProvidedByAndScopeTest.java [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+      return this == obj;
+    }
+  }
+
+  @Test
+  public void testMixClass() throws Exception {
+    final Injector injector =
+        Guice.createInjector(
+            new AbstractModule() {
```
Comment (ogregoire) [inline core/test/com/google/inject/ProvidedByAndScopeTest.java:45]: You can just call `Injector injector = Guice.createInjector();` without any module for the same value.

File: core/test/com/google/inject/ProvidedByAndScopeTest.java (hunk 1 above)
Comment (birdmanmandbir) [reply core/test/com/google/inject/ProvidedByAndScopeTest.java:36]: OK，I will update it today : ) 
Thanks for your advice!

File: core/test/com/google/inject/ProvidedByAndScopeTest.java (hunk 2 above)
Comment (birdmanmandbir) [reply core/test/com/google/inject/ProvidedByAndScopeTest.java:45]: OK，I will update it today : ) 
Thanks for your advice!

**Pull request closed without merging**